### PR TITLE
Add docker

### PR DESCRIPTION
**Pull request by MatMoore**
This creates a consistent development environment which avoids the need for setting up ruby, node, postgres etc.

This is partly based on this guide: https://www.digitalocean.com/community/tutorials/containerizing-a-ruby-on-rails-application-for-development-with-docker-compose

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,57 +23,74 @@ The stack is:
...
+
 ```
 rails spec
 ```
 
+(`docker-compose exec app rails spec` if using docker)
+
+## Editor
+
+You can use your editor of choice but please make sure to reformat your code with [Prettier](https://www.moeckernkiez.de/quartier-moeckernkiez/freie-wohnungen/kosten/) according to the `.prettierrc` file.
```
Comment (herrherrmann) [inline README.md:92]: Wtf, I think back then I copy-pasted the wrong link URL. :D 

```suggestion
You can use your editor of choice but please make sure to reformat your code with [Prettier](https://prettier.io) according to the `.prettierrc` file.
```

**Comment by MatMoore** [comment]
Ah... I think it is successfully creating the dev db but failing on the test one. It should be safe to ignore for running the server, but I expect the tests aren't working properly. I'll try and look into this later in the week.

**Comment by MatMoore** [comment]
By the way, I also though about using the same docker image on CI, but I gave up on that idea because I wasn't sure how to implement it.

I find github actions a bit confusing because there are containers running in containers, and if I built and ran this docker image inside a job I'd have 3 layers of containers 😱

**Comment by herrherrmann** [comment]
We can probably merge this now already, right? Or is there anything still missing?

File: README.md [hunk 2]
```diff
@@ -23,57 +23,74 @@ The stack is:
...
+```
+docker-compose exec app rails db:setup
+docker-compose exec app rails db:migrate
+```
+
+Whenever any of the dependencies change, run `docker-compose build` to rebuild the docker containers.
+
+### Method 2: install all the things
+
+Prerequisites: Ruby 2.6.5, Node >=10.13.0, yarn, postgreSQL 12.2.
```
Comment (herrherrmann) [inline README.md:46]: I'd probably turn this into a list to make it more scannable. And links could also help!

```suggestion
Prerequisites: 

- Ruby 2.6.5 (we recommend using a Ruby version manager like [rbenv](https://github.com/rbenv/rbenv))
- [Node.js](https://nodejs.org) >= 10.13.0
- [Yarn](https://yarnpkg.com)
- PostgreSQL 12.2
```

File: README.md [hunk 3]
```diff
@@ -23,57 +23,74 @@ The stack is:
...
+
 ```
 rails spec
 ```
 
+(`docker-compose exec app rails spec` if using docker)
+
+## Editor
+
+You can use your editor of choice but please make sure to reformat your code with [Prettier](https://prettier.io) according to the `.prettierrc` file.
```
Comment (herrherrmann) [inline README.md:92]: Do you actually use anything for formatting Ruby code? Or a linter/checker? Could be nice to mention it here.

Comment (MatMoore) [reply README.md:97]: Not yet - I've added an issue for it here https://github.com/COVID-science-engine/crowdsourced-papers-frontend/issues/41

**Pull request merged**